core/org.osate.xtext.aadl2/src/org/osate/xtext/aadl2/validation/Aadl2JavaValidator.java

```diff
@@ -3425,7 +3425,8 @@ private void checkFeaturesOfExtendedAbstractType(ComponentType componentType) {
 		for (FeatureType featureType : typesOfInheritedFeatures) {
 			if (!acceptableFeatureTypes.contains(featureType)) {
 
-				String changeFrom = typeCategory.getName();
+				String changeFrom = typeCategory != null ? typeCategory.getName()
```

Comment:
I think this can be resolved in a better way. A few lines earlier is a bunch of if-else-ifs that perform `instanceof` checks and assign values to `typeCategory`. The if-else-ifs should be exhaustive. I think it would be better to add a last else block and throw an `AssertionError` complaining that there was an unrecognized `ComponentType`.